subprojects/plugin-use/src/integTest/groovy/org/gradle/plugin/use/resolve/service/ScriptPluginUseIntegrationTest.groovy

```diff
@@ -0,0 +1,412 @@
+
+        then:
+        server.stop()
+
+        when:
+        succeeds "help", "--offline"
+
+        then:
+        output.contains("Hello from the other side")
+
+        and:
+        operations.hasOperation("Apply script plugin '$scriptUrl' to root project 'root'")
+    }
+
+    def "project script can request multiple script plugins"() {
+
+        given:
+        file("gradle/hello.gradle") << "println('Hello from the other side')"
+        def bye = file("gradle/bye.gradle") << "println('Bye from the other side')"
+
+        and:
+        server.expectGet("/bye.gradle", bye)
+        server.start()
+
+        and:
+        buildFile << """
+
+            plugins {
+                script("gradle/hello.gradle")
+                script("http://localhost:${server.port}/bye.gradle")
+            }
+
+        """.stripIndent()
+
+        when:
+        succeeds "help"
+
+        then:
+        output.contains("Hello from the other side")
+        output.contains("Bye from the other side")
+
+        and:
+        operations.hasOperation("Apply script plugin 'gradle/hello.gradle' to root project 'root'")
+        operations.hasOperation("Apply script plugin 'http://localhost:${server.port}/bye.gradle' to root project 'root'")
+    }
+
+    def "project scripts in a multi-projects build can request script plugins"() {
+
+        given:
+        file("gradle/hello.gradle") << "println(\"Hello from the other side of \$name\")"
+
+        and:
+        settingsFile << """
+
+            rootProject.name = "root"
+            include "a", "b"
+
+        """.stripIndent()
+
+        and:
+        buildFile << """
+
+            plugins {
+                script("gradle/hello.gradle")
+            }
+
+        """.stripIndent()
+        file("a/build.gradle") << """
+
+            plugins {
+                script("../gradle/hello.gradle")
+            }
+
+        """.stripIndent()
+        file("b/build.gradle") << """
+
+            plugins {
+                script("../gradle/hello.gradle")
+            }
+
+        """.stripIndent()
+
+        when:
+        succeeds "help"
+
+        then:
+        output.contains("Hello from the other side of root")
+        output.contains("Hello from the other side of a")
+        output.contains("Hello from the other side of b")
+
+        and:
+        operations.hasOperation("Apply script plugin 'gradle/hello.gradle' to root project 'root'")
+        operations.hasOperation("Apply script plugin 'gradle/hello.gradle' to project ':a'")
+        operations.hasOperation("Apply script plugin 'gradle/hello.gradle' to project ':b'")
+    }
+
+    def "same script plugin requested from different paths gets applied only once"() {
+
+        given:
+        file("gradle/hello.gradle") << "println('Hello from the other side')"
+
+        and:
+        buildFile << """
+
+            plugins {
+                script("gradle/hello.gradle")
+                script("gradle/../gradle/hello.gradle")
```

Comment:
this should really fail given that we disallow duplicate plugin requests, as this two requests for the same thing.

Reply:
It's like using two different ids to refer to the same plugin class. The requests are different but point at the same thing.